perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -606,7 +662,10 @@ protected void removeDirectMember(PerunSession sess, Group group, Member member)
 		// check all relations with this group and call processRelationMembers to reflect changes of removing member from group
 		List<Integer> relations = groupsManagerImpl.getResultGroupsIds(sess, group.getId());
 		for (Integer groupId : relations) {
-			processRelationMembers(sess, groupsManagerImpl.getGroupById(sess, groupId), Collections.singletonList(member), group.getId(), false);
+			Group relationGroup =  groupsManagerImpl.getGroupById(sess, groupId);
+			//Create lock for affected group and member
+			PerunLocksUtils.lockGroupMembership(relationGroup, Arrays.asList(member));
```

Comment:
We might want to check javadoc of ``Collections.singletonList(member)`` used in a code for group relations and compare it with ``Arrays.asList(member)`` if it has any benefits. Probably depends on later usage. Second method is backed by fixed sized array, hence we can't add more items to the list.